src/popsift/common/excl_blk_prefix_sum.h

```diff
@@ -133,16 +132,15 @@ class Block
                 _mapping_writer.set( ebs, self, cell );
             }
 
+            // Wait to update loop_total until everyone is done.
+            __syncthreads();
             if( threadIdx.y == 0 && threadIdx.x == 31 ) {
                 loop_total += ibs;
             }
-            __syncthreads();
         }
 
-        // if( threadIdx.y == 0 && threadIdx.x == 31 )
-        if( threadIdx.y == 0 )
+        if( threadIdx.y == 0 && threadIdx.x == 31 )
```

Comment:
Actually, I think that TotalWriter::set should not be thread-conditional at all.
If the TotalWriter::set function wants to write it in a single thread, it is possible to implement it inside TotalWriter::set - which must be supplied by the caller.
But ExtremaTot in s_orientation.cu is actually writing into the thread-specific variable total_ori (local variable in ori_prefix_sum). With a thread-conditional call to TotalWriter::set, most threads have an uninitialized total_ori, and it works only because total_ori is never used for anything.

Reply:
Agreed - condition was removed via `6e4428b`.